snuba/query/parser/__init__.py

```diff
@@ -373,6 +374,38 @@ def _expand_aliases(query: Query) -> None:
 ARRAYJOIN_FUNCTION_MATCH = FunctionCallMatch(StringMatch("arrayJoin"), None)
 
 
+def _mangle_aliases(query: Query) -> None:
+    alias_prefix = "_snuba_"
+
+    def transform_alias(alias: Optional[str]) -> Optional[str]:
+        if alias is not None:
+            return f"{alias_prefix}{alias}"
+
+        return alias
+
+    query.transform_expressions(
+        lambda expr: replace(expr, alias=transform_alias(expr.alias))
+    )
+
+    for expression in query.get_all_expressions():
+        if expression.alias:
+            expression.transform(
```

Comment:
Expressions are not transformed in place. This method will return a new expression object that is being ignored here.